ui/src/app/testing/fake-aggregation.service.ts

```diff
@@ -89,9 +91,13 @@ const aggregationResponse: AggregationResponse =
    }
   };
 
+// This is the number of mat-card that would display on dashboard page,
+// remember to change the number if you change the aggregationResponse above
+export const CARD_NUM = 4;
```

Comment:
I think it would probably be slightly clearer to get rid of this property and instead do this from the test to avoid the risk of these going out of sync:

`TEST_AGGREGATION_RESPONSE.aggregations.length`

Also nitpick, but w.r.t. separation of concerns this service stub shouldn't know about "cards" since it's a presentation concern. The dashboard component renders specific card components in the UI, but the AggregationService should only know about the details of the API. This service could potentially be used elsewhere in the app in the future where there's no card UX, or for example in the future we might only display the first N cards.

Reply:
Adapted!